### PR TITLE
Fix a leak in pi_unified_runtime.cpp.

### PR DESCRIPTION
**Pull request by yingcong-wu**
`LoaderConfig` is created and stored in a local pointer and never released when done using, causing it to be leaked.
This patch releases the `LoaderConfig` when finished using it.

### CODE REVIEW

**Review by kbenzie — APPROVED** [review]

File: sycl/plugins/unified_runtime/pi_unified_runtime.cpp
Comment (ldrumm) [inline sycl/plugins/unified_runtime/pi_unified_runtime.cpp:1352]: This is an improvement, but if `urLoaderConfigEnableLayer` or `urLoaderInit` fail we still leak. Do we need to fix this RAII-style instead?

Comment (kbenzie) [reply sycl/plugins/unified_runtime/pi_unified_runtime.cpp:1352]: I did recently add RAII wrappers to the UR CTS, there's deffo a case for making those first class citizens so they can be reused where needed.

Comment (yingcong-wu) [reply sycl/plugins/unified_runtime/pi_unified_runtime.cpp:1352]: I will look into the RAII wrappers. Hi @kbenzie , do you have some links to some documents/commit that I can start with? Thanks.

Comment (yingcong-wu) [reply sycl/plugins/unified_runtime/pi_unified_runtime.cpp:1352]: Hi @kbenzie , did UR alrdeay provide such RAII warppers or did you mean that you plan to add such wrappers to UR?

Comment (kbenzie) [reply sycl/plugins/unified_runtime/pi_unified_runtime.cpp:1352]: We do not currently provide public RAII wrappers, I've created https://github.com/oneapi-src/unified-runtime/issues/1334 to address this.

I don't think this should block this PR.

**Comment by KseniyaTikhomirova** [comment]
@yingcong-wu could you please review the answers & approve if no objections? Would be great to get this fix merged since it causes failures of 9 SYCL tests.

**Comment by yingcong-wu** [comment]
Hi @kbenzie, if the RAII wrapper does not block this PR, could you please help land this patch? Thank you.

**Pull request merged**
